### PR TITLE
Ml 144 improved animations

### PR DESCRIPTION
**Pull request by pkrucz00**
1. Improved animation of floating heart (centred and visible for presenters).
2. Added confetti animation.

### CODE REVIEW

File: lib/membrane_live_web/channels/event_channel.ex [hunk 1]
```diff
@@ -219,8 +219,13 @@ defmodule MembraneLiveWeb.EventChannel do
...
-    broadcast(socket, "animation", %{})
+  def handle_in("heart_reaction", _data, socket) do
+    broadcast(socket, "heart_animation", %{})
+    {:noreply, socket}
+  end
+
+  def handle_in("confetti_reaction", _data, socket) do
+    broadcast(socket, "confetti_animation", %{})
     {:noreply, socket}
   end
```
Comment (Rados13) [inline lib/membrane_live_web/channels/event_channel.ex:230]: What do you think about a change like this?
```suggestion
  def handle_in(animation_type <> "_reaction", _data, socket) when animation_type in ["confetti", "heart"] do
    broadcast(socket,animation_type  <>"_animation", %{})
    {:noreply, socket}
  end
```

**Review by Rados13 — DISMISSED** [review]
The code looks good it can go to QA

File: lib/membrane_live_web/channels/event_channel.ex (hunk 1 above)
Comment (pkrucz00) [reply lib/membrane_live_web/channels/event_channel.ex:230]: Looks great! I'll commit it right away

Comment (pkrucz00) [reply lib/membrane_live_web/channels/event_channel.ex]: Although the code looks good, it does not work. The `<>` operator needs to have an explicit byte_size of all the arguments excluding the last one. Fixed that by switching animation type with `"reaction"` (it's `reaction_heart` now, not `heart_reaction`).

File: assets/js/components/event/ModePanel.tsx [hunk 2]
```diff
@@ -31,23 +32,36 @@ type ModePanelProps = {
...
+    };
+  }, [isHeartClicked]);
+
+  useEffect(() => {
+    const ref = setTimeout(() => setConfettiClicked(false), 2_500);
     return () => {
       clearTimeout(ref);
     };
-  }, [isClicked]);
+  }, [isConfettiClicked]);
```
Comment (kamil-stasiak) [inline assets/js/components/event/ModePanel.tsx:55]: You could create custom hook for that. It toggles value for some time.

File: assets/js/components/event/animations/ConfettiAnimation.tsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    const ref = eventChannel?.on("animation_confetti", () => {
+      setIsRun(true);
+      timeoutRef = setTimeout(() => setIsRun(false), 2_500);
+    });
+
+    return () => {
+      timeoutRef && clearTimeout(timeoutRef);
+      eventChannel?.off("animation_confetti", ref);
+    };
+  }, [eventChannel]);
```
Comment (kamil-stasiak) [inline assets/js/components/event/animations/ConfettiAnimation.tsx:24]: maybe it is possible to include this logic in previously suggested hook

File: assets/js/components/event/animations/HeartAnimation.tsx [hunk 4]
```diff
@@ -80,23 +80,23 @@ const AnimationComponent = ({ eventChannel }: HeartAnimationProps) => {
...
+    const randomXFactor = Math.random() * 200 - 100;
     const randomYFactor = Math.random() * 20;
 
-    return generateHeart(200 + cursorXOffset + randomXFactor, 10 + cursorYOffset + randomYFactor, bound, start, scale);
+    return generateHeart(cursorXOffset + randomXFactor, 10 + cursorYOffset + randomYFactor, bound, start, scale);
   }, [generateHeart]);
 
   useEffect(() => {
-    const ref = eventChannel?.on("animation", () => {
+    const ref = eventChannel?.on("animation_heart", () => {
```
Comment (kamil-stasiak) [inline assets/js/components/event/animations/HeartAnimation.tsx:91]: every messageId could be constant

**Review by kamil-stasiak — APPROVED** [review]

**Pull request merged**